src/Classes/PolygonMask.ts

```diff
@@ -0,0 +1,222 @@
+                    const xj = this.raw[j][0], yj = this.raw[j][1];
+                    const intersect = ((yi > y) != (yj > y)) && (x < (xj - xi) * (y - yi) / (yj - yi) + xi);
+
+                    if (intersect) {
+                        inside = !inside;
+                    }
+                }
+
+                if (options.polygon.type === PolygonMaskType.inside) {
+                    return inside;
+                } else if (options.polygon.type === PolygonMaskType.outside) {
+                    return !inside;
+                }
+            } else {
+                console.error('No polygon found, you need to specify SVG url in config.');
+                return true;
+            }
+        } else {
+            return true;
+        }
+    }
+
+    public randomPointInPolygon(): ICoordinates {
+        const container = this.container;
+        const position = {
+            x: Math.random() * container.canvas.width,
+            y: Math.random() * container.canvas.height,
+        };
+
+        if (this.checkInsidePolygon(position)) {
+            return position;
+        } else {
+            return this.randomPointInPolygon();
+        }
+    }
+
+    /**
+     * Depends on SVGPathSeg API polyfill https://github.com/progers/pathseg for Chrome
+     * Deprecate SVGPathElement.getPathSegAtLength removed in:
+     * Chrome for desktop release 62
+     * Chrome for Android release 62
+     * Android WebView release 62
+     * Opera release 49
+     * Opera for Android release 49
+     */
+    public async parseSvgPathToPolygon(svgUrl?: string) {
+        const container = this.container;
+        const options = container.options;
+
+        svgUrl = svgUrl || options.polygon.url;
+        // Load SVG from file on server
+        if (!this.path || !this.svg) {
+            const req = await fetch(svgUrl);
+            if (req.ok) {
+                const xml = await req.text();
+
+                const doc = new XMLDocument();
+
+                doc.write(xml);
+
+                this.svg = doc.getElementsByTagName("svg")[0];
+                this.path = doc.getElementsByTagName("path")[0];
+            } else {
+                console.error("tsParticles Error - during polygon mask download");
+                return;
+            }
+        }
+
+        this.width = parseFloat(this.svg.getAttribute("width") || "0");
+        this.height = parseFloat(this.svg.getAttribute("height") || "0");
+
+        /* centering of the polygon mask */
+        this.offset = {
+            x: container.canvas.width / 2 - this.width / 2,
+            y: container.canvas.height / 2 - this.height / 2
+        };
+
+
+        const len = this.path.pathSegList.numberOfItems;
+        const polygonRaw = [];
+        const p = {
+            x: 0,
+            y: 0
+        };
+
+        for (let i = 0; i < len; i++) {
+            const pathSeg: any = this.path.pathSegList.getItem(i);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5042590)